### PR TITLE
0.3 : add resetDelay prop + add componentWillReceiveProps()

### PR DESCRIPTION
**Pull request by guillaumearm**
- add resetDelay prop : time in ms you have before reset N. (default to 1500)
- add componentWillReceiveProps() : 
  - refresh timeout when resetDelay prop change.
  - reset N when konami code change.

### CODE REVIEW

**Comment by FloValence** [comment]
Hello @guillaumearm,
Thank you for your pull request, I have a question:
What is the advantage for the user with this delay management? I mean as it resets `n` for each wrong key I don't think that we need to set a timeout. It adds a lot of logic.
What do you think?

**Comment by guillaumearm** [comment]
Hi,
I think a konami code should be typed fast enough.
but it's true, it adds a lot of logic, so i let you decide about this PR ;-)

**Comment by FloValence** [comment]
Hello,
Can you make it optional ?
And I think 1500ms might be too quick as a default value. Maybe 3000?
And then I will merge it.

Thank you

**Comment by guillaumearm** [comment]
really, you should try it, 1500ms is perfect ;-) it is 1500ms between keys typed.

and yes i will make it optional. what do you think to put resetDelay to 0 for disabling this effect ?

Anyway, we need to document this behavior before merging ;-)

**Comment by FloValence** [comment]
I'll try ^^
Yes setting resetDelay to 0 should disable the timeout.
+1 for the documentation :)

**Comment by guillaumearm** [comment]
done.
So last commit to set to 2000ms ?

File: Konami.js [hunk 1]
```diff
@@ -6,26 +6,54 @@ class Konami extends Component {
...
+    this.delayOff()
   }
 
+  delayOff() {
+      if (this.delayId) clearTimeout(this.delayId)
+  }
+
+  delayOn() {
+     this.delayOff()
+     if (this.props.resetDelay > 0)
```
Comment (FloValence) [inline Konami.js]: I would have done something like that, to leave early

```
if (this.props.resetDelay <= 0) return
this.delayOff()
this.delayId = setTimeout(() => this.resetN(), this.props.resetDelay)
```

**Comment by FloValence** [comment]
Yes, please 2000ms should be good, I still feel 1500ms too short :)

**Comment by guillaumearm** [comment]
that's all ;-)
need to commit new build ? 
or waiting to commit hook as said in #7  ?

File: Konami.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ class Konami extends Component {
     const resetDelayHasChanged = resetDelay !== this.props.resetDelay
     if (resetDelayHasChanged && resetDelay <= 0) {
       this.delayOff()
-    else if (resetDelayHasChanged) {
+    } else if (resetDelayHasChanged) {
```
Comment (guillaumearm) [inline Konami.js:22]: sorry about that.

**Comment by FloValence** [comment]
Yes, build it please, I didn't have the time to work the hook yet.
By the end of week, hopefully :-)

**Pull request merged**